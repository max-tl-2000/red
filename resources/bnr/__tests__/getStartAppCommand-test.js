/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

/* eslint-disable global-require */
const { mockModules } = require('test-helpers/mocker').default(jest);

describe('getStartAppCommand', () => {
  let getStartAppCommand;

  beforeEach(() => {
    jest.resetModules();

    mockModules({
      path: {
        resolve: p => p,
      },
    });

    getStartAppCommand = require('../getStartAppCommand').default;
  });

  it('it should create a command to execute the app in development mode by default', () => {
    const fn = getStartAppCommand('./bin/server', {
      NODE_PATH: './server',
      RED_PROCESS_NAME: 'leasing-server',
      PORT: 3000,
      API_PORT: 3030,
    });
    expect(fn({ args: {} })).toMatchSnapshot();
  });

  it('if production is set the NODE_ENV should be production', () => {
    const fn = getStartAppCommand('./bin/server', {
      NODE_PATH: './server',
      RED_PROCESS_NAME: 'leasing-server',
      PORT: 3000,
      API_PORT: 3030,
    });
    expect(fn({ args: { production: true } })).toMatchSnapshot();
  });
  it('if production is set the NODE_ENV should be integration', () => {
    const fn = getStartAppCommand('./bin/server', {
      NODE_PATH: './server',
      RED_PROCESS_NAME: 'leasing-server',
      PORT: 3000,
      API_PORT: 3030,
    });
    expect(fn({ args: { integration: true } })).toMatchSnapshot();
  });

  it('should set RED_LOGGER_STDOUT to false in case quiet option is set', () => {
    const fn = getStartAppCommand('./bin/server', {
      NODE_PATH: './server',
      RED_PROCESS_NAME: 'leasing-server',
      PORT: 3000,
      API_PORT: 3030,
    });
    expect(fn({ args: { quiet: true } })).toMatchSnapshot();
  });

  it('should RED_LOGGER_STDOUT also when production is true even if quiet was specified', () => {
    const fn = getStartAppCommand('./bin/server', {
      NODE_PATH: './server',
      RED_PROCESS_NAME: 'leasing-server',
      PORT: 3000,
      API_PORT: 3030,
    });
    expect(fn({ args: { quiet: true, production: true } })).toMatchSnapshot();
  });

  it('should create a command to start an app in debug mode', () => {
    const fn = getStartAppCommand('./bin/server', {
      NODE_PATH: './server',
      RED_PROCESS_NAME: 'leasing-server',
      PORT: 3000,
      API_PORT: 3030,
    });
    expect(fn({ args: { quiet: true, debug: true } })).toMatchSnapshot();
  });

  it('should create a command to start an app in debug mode with debugBrk set if required as well as a different port', () => {
    const fn = getStartAppCommand('./bin/server', {
      NODE_PATH: './server',
      RED_PROCESS_NAME: 'leasing-server',
      PORT: 3000,
      API_PORT: 3030,
    });
    expect(fn({ args: { quiet: true, debug: true, debugPort: 9888, debugBrk: true } })).toMatchSnapshot();
  });
});
