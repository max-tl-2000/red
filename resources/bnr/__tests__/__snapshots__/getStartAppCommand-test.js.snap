// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStartAppCommand if production is set the NODE_ENV should be integration 1`] = `
Object {
  "cmd": "node --require ./enable-ts.js ./common/server/forever.js ./bin/server",
  "env": Object {
    "API_PORT": 3030,
    "BABEL_CACHE_PATH": "node_modules/.cache/babel/leasing-server",
    "NODE_ENV": "integration",
    "NODE_PATH": "./server",
    "PORT": 3000,
    "RED_LOGGER_USE_STDOUT": false,
    "RED_PROCESS_NAME": "leasing-server",
    "TZ": "UTC",
  },
}
`;

exports[`getStartAppCommand if production is set the NODE_ENV should be production 1`] = `
Object {
  "cmd": "node --require ./enable-ts.js ./common/server/forever.js ./bin/server",
  "env": Object {
    "API_PORT": 3030,
    "BABEL_CACHE_PATH": "node_modules/.cache/babel/leasing-server",
    "NODE_ENV": "production",
    "NODE_PATH": "./server",
    "PORT": 3000,
    "RED_LOGGER_USE_STDOUT": false,
    "RED_PROCESS_NAME": "leasing-server",
    "TZ": "UTC",
  },
}
`;

exports[`getStartAppCommand it should create a command to execute the app in development mode by default 1`] = `
Object {
  "cmd": "babel-node --extensions '.ts,.js,.json'  ./bin/server",
  "env": Object {
    "API_PORT": 3030,
    "BABEL_CACHE_PATH": "node_modules/.cache/babel/leasing-server",
    "NODE_ENV": "development",
    "NODE_PATH": "./server",
    "PORT": 3000,
    "RED_LOGGER_USE_STDOUT": true,
    "RED_PROCESS_NAME": "leasing-server",
    "TZ": "UTC",
  },
}
`;

exports[`getStartAppCommand should RED_LOGGER_STDOUT also when production is true even if quiet was specified 1`] = `
Object {
  "cmd": "node --require ./enable-ts.js ./common/server/forever.js ./bin/server",
  "env": Object {
    "API_PORT": 3030,
    "BABEL_CACHE_PATH": "node_modules/.cache/babel/leasing-server",
    "NODE_ENV": "production",
    "NODE_PATH": "./server",
    "PORT": 3000,
    "RED_LOGGER_USE_STDOUT": false,
    "RED_PROCESS_NAME": "leasing-server",
    "TZ": "UTC",
  },
}
`;

exports[`getStartAppCommand should create a command to start an app in debug mode 1`] = `
Object {
  "cmd": "babel-node --extensions '.ts,.js,.json' --inspect=9229 ./bin/server",
  "env": Object {
    "API_PORT": 3030,
    "BABEL_CACHE_PATH": "node_modules/.cache/babel/leasing-server",
    "DEBUG": true,
    "NODE_ENV": "development",
    "NODE_PATH": "./server",
    "PORT": 3000,
    "RED_LOGGER_USE_STDOUT": false,
    "RED_PROCESS_NAME": "leasing-server",
    "TZ": "UTC",
  },
}
`;

exports[`getStartAppCommand should create a command to start an app in debug mode with debugBrk set if required as well as a different port 1`] = `
Object {
  "cmd": "babel-node --extensions '.ts,.js,.json' --inspect=9888 --debug-brk ./bin/server",
  "env": Object {
    "API_PORT": 3030,
    "BABEL_CACHE_PATH": "node_modules/.cache/babel/leasing-server",
    "DEBUG": true,
    "NODE_ENV": "development",
    "NODE_PATH": "./server",
    "PORT": 3000,
    "RED_LOGGER_USE_STDOUT": false,
    "RED_PROCESS_NAME": "leasing-server",
    "TZ": "UTC",
  },
}
`;

exports[`getStartAppCommand should set RED_LOGGER_STDOUT to false in case quiet option is set 1`] = `
Object {
  "cmd": "babel-node --extensions '.ts,.js,.json'  ./bin/server",
  "env": Object {
    "API_PORT": 3030,
    "BABEL_CACHE_PATH": "node_modules/.cache/babel/leasing-server",
    "NODE_ENV": "development",
    "NODE_PATH": "./server",
    "PORT": 3000,
    "RED_LOGGER_USE_STDOUT": false,
    "RED_PROCESS_NAME": "leasing-server",
    "TZ": "UTC",
  },
}
`;
